### PR TITLE
feat(infra): add check-warp-deply script that checks all AW managed w…

### PR DESCRIPTION
**Pull request by Mo-Hussain**
### Description

- add a script that allows use to check the warp route deployments for all AW managed warp routes

### Drive-by changes

- split up check-deploy script into reusable method 
- move check-deploy into check directory in infra

### Testing

Manual

### CODE REVIEW

File: typescript/infra/fork.sh [hunk 1]
```diff
@@ -39,7 +39,7 @@ execute_command() {
 }
 
 echo "Checking deploy"
-execute_command "yarn tsx ./scripts/check-deploy.ts -e $ENVIRONMENT -f $CHAIN -m $MODULE"
+execute_command "yarn tsx ./scripts/check/check-deploy.ts -e $ENVIRONMENT -f $CHAIN -m $MODULE"
```
Comment (Mo-Hussain) [inline typescript/infra/fork.sh:42]: Would we need to do this anywhere else where check-deploy is called

File: typescript/infra/scripts/check/check-utils.ts [hunk 2]
```diff
@@ -11,47 +15,52 @@ import {
...
+  return withAsDeployer(
+    withGovern(withChain(withFork(withContext(getRootArgs())))),
+  );
+}
 
-function getArgs() {
-  return withChain(
-    withWarpRouteId(withModuleAndFork(withContext(getRootArgs()))),
-  )
-    .boolean('asDeployer')
```
Comment (Mo-Hussain) [inline typescript/infra/scripts/check/check-utils.ts:47]: Pull these into helpers

File: typescript/infra/scripts/check/check-utils.ts [hunk 3]
```diff
@@ -11,47 +15,52 @@ import {
...
+  withModule,
   withWarpRouteId,
-} from './agent-utils.js';
-import { getEnvironmentConfig, getHyperlaneCore } from './core-utils.js';
-import { getHelloWorldApp } from './helloworld/utils.js';
+} from '../agent-utils.js';
+import { getEnvironmentConfig, getHyperlaneCore } from '../core-utils.js';
+import { getHelloWorldApp } from '../helloworld/utils.js';
+
+export function getCheckArgs() {
```
Comment (Mo-Hussain) [inline typescript/infra/scripts/check/check-utils.ts:51]: Use this in `check-warp-deploy` script

File: typescript/infra/scripts/check/check-utils.ts [hunk 4]
```diff
@@ -214,56 +299,10 @@ async function check() {
     );
     governor = new ProxiedRouterGovernor(checker, ica);
   } else {
-    console.log(`Skipping ${module}, checker or governor unimplemented`);
+    // TODO: should we throw here instead?
```
Comment (Mo-Hussain) [inline typescript/infra/scripts/check/check-utils.ts:302]: Any objections to throwing here?

File: typescript/infra/scripts/check/check-warp-deploy.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+import { WarpRouteIds } from '../../config/warp.js';
+import { Modules } from '../agent-utils.js';
+
+import { check, getCheckArgs } from './check-utils.js';
+
+async function checkWarp() {
+  const argv = await getCheckArgs().argv;
+
+  // TODO: consider retrying this if check throws an error
```
Comment (Mo-Hussain) [inline typescript/infra/scripts/check/check-warp-deploy.ts:11]: I've seen issues with linea RPCs causing this to fail

File: typescript/infra/scripts/check/check-utils.ts (hunk 4 above)
Comment (paulbalaji) [reply typescript/infra/scripts/check/check-utils.ts:299]: i'm down with throwing

**Review by paulbalaji — APPROVED** [review]

**Pull request merged**